File: app/models/pet.rb
class Pet < ActiveRecord::Base
  belongs_to :owner, :class_name => "User"
  belongs_to :finder, :class_name => "User"

  attr_accessible :cur_location_lat, :cur_location_long, :description, :last_seen_location_lat, :last_seen_location_long,
            :name, :picture, :status, :animal_type, :breed, :sex, :color, :size

  def self.localPets(range = 25, location_lat, location_long)
    begin
      range = range.to_f * 1609
    rescue
      range = 25*1609
    end

    self.find_by_sql(
      "SELECT * FROM pets WHERE " +
      "earth_box(" +
        "ll_to_earth(#{location_lat}, #{location_long}), " +
        "#{range}" +
      ") @> ll_to_earth(last_seen_location_lat, last_seen_location_long)" +
      " AND updated_at > current_timestamp - interval '21' day" +
      " ORDER BY updated_at DESC"
    )
  end

  def self.uploadPicture(uploaded_io)
    begin
      require 'aws/s3'
      AWS::S3::Base.establish_connection!(
        :access_key_id     => AWS_CONFIG['access_key'],
        :secret_access_key => AWS_CONFIG['secret']
      )

      key = SecureRandom.hex() + ".jpg"
      AWS::S3::S3Object.store(key, uploaded_io, 'lostanimals', :content_type => uploaded_io.content_type, :access => :public_read)

      "https://s3.amazonaws.com/lostanimals/" + key
    rescue
      nil
    end
  end
end

File: LostAnimalsProject/LostAnimals/src/main/java/com/r0adkll/lostanimals/server/model/Pet.java
package com.r0adkll.lostanimals.server.model;

import org.json.JSONObject;

import java.io.Serializable;

/**
 * Created by r0adkll on 6/1/13.
 */
public class Pet implements Serializable{


    /**
     * Variables
     */
    public String animal_type;
    public String breed;
    public String color;
    public String created_at;
    public String cur_location_lat;
    public String cur_location_long;
    public String description;
    public String finder_id;
    public int id;
    public double last_seen_location_lat;
    public double last_seen_location_long;
    public String name;
    public String owner_id;
    public String picture;
    public String sex;
    public String size;
    public String status;
    public String updated_at;


    /**
     * JSON Constructor
     * @param json  the json object ot construct from
     */
    public Pet(JSONObject json){
        animal_type = json.optString("animal_type");
        breed = json.optString("breed");
        color = json.optString("color");
        created_at = json.optString("created_at");
        cur_location_lat = json.optString("cur_location_lat");
        cur_location_long = json.optString("cur_location_long");
        description = json.optString("description");
        finder_id = json.optString("finder_id");
        id = json.optInt("id");
        last_seen_location_lat = json.optDouble("last_seen_location_lat");
        last_seen_location_long = json.optDouble("last_seen_location_long");
        name = json.optString("name");
        owner_id = json.optString("owner_id");
        picture = json.optString("picture");
        sex = json.optString("sex");
        size = json.optString("size");
        status = json.optString("status");
        updated_at = json.optString("updated_at");
    }



}
